airflow_versions/http_client.go

```diff
@@ -0,0 +1,70 @@
+package airflowversions
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/astronomer/astro-cli/config"
+	"github.com/astronomer/astro-cli/pkg/httputil"
+	"github.com/pkg/errors"
+)
+
+// Client containers the logger and HTTPClient used to communicate with the HoustonAPI
+type Client struct {
+	HTTPClient *httputil.HTTPClient
+}
+
+// NewClient returns a new Client with the logger and HTTP client setup.
+func NewClient(c *httputil.HTTPClient) *Client {
+	return &Client{
+		HTTPClient: c,
+	}
+}
+
+type Request struct{}
+
+// Do (request) is a wrapper to more easily pass variables to a client.Do request
+func (r *Request) DoWithClient(api *Client) (*Response, error) {
+	doOpts := httputil.DoOptions{
+		Headers: map[string]string{
+			"Accept": "application/json",
+		},
+	}
+
+	return api.Do(doOpts)
+}
+
+// Do (request) is a wrapper to more easily pass variables to a client.Do request
+func (r *Request) Do() (*Response, error) {
+	return r.DoWithClient(NewClient(httputil.NewHTTPClient()))
+}
+
+// Do executes a query against the updates astronomer API, logging out any errors contained in the response object
+func (c *Client) Do(doOpts httputil.DoOptions) (*Response, error) {
```

Comment:
Line 40 and 45, can two func bear the same name in a package?

Reply:
@Simpcyclassy this is OK because both `Do` functions are methods with different receivers.  You can learn more about this in the Golang tour: https://tour.golang.org/methods/1